cmds/core/gosh/completer_liner.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build !tinygo && !plan9 && !goshsmall && goshliner
+// +build !tinygo,!plan9,!goshsmall,goshliner
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/peterh/liner"
+	"mvdan.cc/sh/v3/interp"
+	"mvdan.cc/sh/v3/syntax"
+)
+
+// HistFile is the history file.
+// This might, possibly, use GetPid to avoid gosh'es writing over each other
+var HistFile = filepath.Join(os.TempDir(), "gosh.history")
+
+var completion = flag.Bool("comp", true, "Enable tabcompletion and a more feature rich editline implementation")
+
+func runInteractive(runner *interp.Runner, parser *syntax.Parser, stdout, stderr io.Writer) error {
+	input := liner.NewLiner()
+	defer input.Close()
+
+	f, err := os.Open(HistFile)
+	if err == nil {
+		input.ReadHistory(f)
+	} else if f, err = os.Create(HistFile); err != nil {
```

Comment:
we had this weird problem at one point where elvish was impossible to use if it could not create a history file. I'm wondering if, in the event we can't create a history file for some reason, we shouldn't still proceed. 

IOW, if we can't create the file, could we just keep calm and carry on? I tend to think yes, but tell them what happened.

Reply:
yeah, that's fair, and done.